cmd/nerdctl/run.go

```diff
@@ -432,7 +432,28 @@ func runAction(clicontext *cli.Context) error {
 		}
 
 		resolvConfPath := filepath.Join(stateDir, "resolv.conf")
-		if err := dnsutil.WriteResolvConfFile(resolvConfPath, strutil.DedupeStrSlice(clicontext.StringSlice("dns"))); err != nil {
+		conf, err := resolvconf.Get()
+		if err != nil {
+			return err
+		}
+		slirp4Dns := []string{}
+		if rootlessutil.IsRootlessChild() {
+			slirp4Dns, err = dnsutil.GetSlirp4netnsDns()
+			if err != nil {
+				return err
+			}
+		}
```

Comment:
Thanks for working on this, but I guess we do not need to detect the slirp4netns DNS.

We can just use `/etc/resolv.conf` (`/run/systemd/resolve/resolv.conf`) for both rootful and rootless.

Reply:
Indeed, in case of rootless we run in the mount ns of rootlesskit thus having the /etc/resolv.conf containing the slirp dns right ?

Reply:
Hmm, I guess we can merge this right now and revisit later.